wweval/R/ms_fig3.R

```diff
@@ -0,0 +1,266 @@
+      show.legend = FALSE
+    ) +
+    geom_line(
+      data = hosp |> dplyr::filter(model_type == "ww"),
+      aes(
+        x = date, y = `0.5`, group = forecast_date
+      ),
+      color = "cornflowerblue"
+    ) +
+    geom_ribbon(
+      data = hosp |> dplyr::filter(model_type == "ww"),
+      aes(
+        x = date, ymin = `0.025`, ymax = `0.975`,
+        group = as.character(forecast_date)
+      ), alpha = 0.1,
+      fill = "cornflowerblue",
+      show.legend = FALSE
+    ) +
+    geom_ribbon(
+      data = hosp |> dplyr::filter(model_type == "ww"),
+      aes(
+        x = date, ymin = `0.25`, ymax = `0.75`,
+        group = as.factor(forecast_date)
+      ),
+      alpha = 0.1,
+      fill = "cornflowerblue",
+      show.legend = FALSE
+    ) +
+    geom_line(
+      data = hosp |> dplyr::filter(model_type == "hosp"),
+      aes(
+        x = date, y = `0.5`, group = forecast_date
+      ),
+      color = "purple4"
+    ) +
+    geom_ribbon(
+      data = hosp |> dplyr::filter(model_type == "hosp"),
+      aes(
+        x = date, ymin = `0.025`, ymax = `0.975`,
+        group = as.character(forecast_date)
+      ), alpha = 0.1,
+      fill = "purple4",
+      show.legend = FALSE
+    ) +
+    geom_ribbon(
+      data = hosp |> dplyr::filter(model_type == "ww"),
+      aes(
+        x = date, ymin = `0.25`, ymax = `0.75`,
+        group = as.factor(forecast_date)
+      ),
+      fill = "purple4",
+      alpha = 0.1,
+      show.legend = FALSE
+    ) +
+    xlab("") +
+    ylab("Daily hospital admissions") +
+    ggtitle(glue::glue(
+      "{horizon_to_plot}"
+    )) +
+    theme_bw() +
+    scale_color_discrete() +
+    scale_fill_discrete() +
+    scale_x_date(
+      date_breaks = "2 weeks",
+      labels = scales::date_format("%Y-%m-%d")
+    ) +
+    theme_bw() +
+    theme(
+      axis.text.x = element_text(
+        size = 8, vjust = 1,
+        hjust = 1, angle = 45
+      ),
+      axis.title.x = element_text(size = 12),
+      axis.title.y = element_text(size = 12),
+      plot.title = element_text(
+        size = 10,
+        vjust = 0.5, hjust = 0.5
+      )
+    )
+
+  return(p)
+}
+
+#' Make CRPS underlay figure
+#'
+#' @param scores A tibble of scores by location, forecast date, date and model,
+#' the ouput of `scoringutils::score()` on samples.
+#' @param loc_to_plot A  string indicating the state abbreviations of the state
+#' to plot
+#' @param horizon_to_plot A string indicating what horizon period to plot,
+#' one of `nowcast`, `1 wk`, or `4 wks`
+#'
+#' @return A ggplot object containing a bar chart of the crps score averaged
+#' across the horizon for each forecast date, colored by the model type
+#' @export
+make_fig3_crps_underlay_fig <- function(scores,
+                                        loc_to_plot,
+                                        horizon_to_plot) {
+  scores_by_horizon <- scores |>
+    dplyr::filter(location == loc_to_plot) |>
+    data.table::as.data.table() |>
```

Comment:
is `scores` a `tibble` or a `data.table`? docs above say it's a tibble but also that it's the output of `scoringutils::score()`, which returns a `data.table`.

Assuming it's a `data.table`. You could use `data.table`'s own filtering methods and then you don't need to coerce it back to a `data.table` in the next step. Either way, need to be careful about evaluating `loc_to_plot` to avoid [column name clashes](https://stackoverflow.com/a/57856144), though:

`data.table`:
```suggestion
  scores_by_horizon <- scores[location == get("loc_to_plot", 1L), ] |>
```

`dplyr`:
```suggestion
  scores_by_horizon <- scores |>
    dplyr::filter(location == !!loc_to_plot) |>
    data.table::as.data.table() |>
```

Reply:
I will adjust the documentation -- it is a tibble, it is the output of `scoringutils()` that I then transform back into a tibble bc I for the most part don't like using `data.table()`.